### PR TITLE
New audio peak limiter library.

### PR DESCRIPTION
**Pull request by kencooke**
Fast lookahead log-domain peak limiter intended to replace all hard-clipping in the audio pipeline.  Converts float audio input (where level may exceed 0dBFS) to 16-bit audio output using very high quality dynamic-range compression and dithering.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 50437307833ddc953d329612b973a9e556117f53 available at: https://gist.github.com/c52aee3817c0008cef1e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6d9fe967da84cd00a0e8eacb1b1033d928756f56 available at: https://gist.github.com/c52aee3817c0008cef1e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 90bf3989783aead4b6965be8eb6f210f055285b9 available at: https://gist.github.com/c52aee3817c0008cef1e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 261f5c45de459ea4a4433e7f9420d4e773ae31e1 available at: https://gist.github.com/c52aee3817c0008cef1e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b346f36b898ae67d6c11831099ac179217333586 available at: https://gist.github.com/c52aee3817c0008cef1e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 06bde077b262b68d377e75084abdd37692468da8 available at: https://gist.github.com/c52aee3817c0008cef1e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.dmg

File: libraries/audio/src/AudioLimiter.cpp [hunk 1]
```diff
@@ -0,0 +1,712 @@
...
+
+    // absolute value
+    int32_t peak = u & 0x7fffffff;
+
+    // split into e and x - 1.0
+    int32_t e = IEEE754_EXPN_BIAS - (peak >> IEEE754_MANT_BITS) + LOG2_HEADROOM;
+    int32_t x = (peak << (31 - IEEE754_MANT_BITS)) & 0x7fffffff;
+
+    // saturate
+    if (e > 31) return 0x7fffffff;
```
Comment (ZappoMan) [inline libraries/audio/src/AudioLimiter.cpp]: coding standard brace placement:
```
if (e > 31) {
    return 0x7fffffff;
}
```
Coding standard - magic numbers:  What is `0x7fffffff`? Is it a meaningful number or is it a semantic code to indicate something? I assume it's the mask to rip off the top level bit... so maybe we should give it a name.

**Comment by ZappoMan** [comment]
I added QA Approved to this, since this is a library not yet being called, so it has no apparent effect on the interface or domain at this point.

There's a couple small coding standard issues, which when fixed, I'd gladly call dev approved and we can merge it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d390c60ef799339fe136d0e82a3cb262c5867ef8 available at: https://gist.github.com/c52aee3817c0008cef1e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7229.dmg

**Pull request merged**